xbmc/powermanagement/PowerManager.cpp

```diff
@@ -246,16 +247,28 @@ void CPowerManager::OnSleep()
   if (dialog)
     dialog->Open();
 
-  CLog::Log(LOGNOTICE, "%s: Running sleep jobs", __FUNCTION__);
+  CLog::Log(LOGNOTICE, "CPowerManager::OnSleep(): Running sleep jobs");
 
   // stop lirc
 #if defined(HAS_LIRC) || defined(HAS_IRSERVERSUITE)
-  CLog::Log(LOGNOTICE, "%s: Stopping lirc", __FUNCTION__);
+  CLog::Log(LOGNOTICE, "CPowerManager::OnSleep(): Stopping lirc");
   CBuiltins::GetInstance().Execute("LIRC.Stop");
 #endif
 
   CServiceBroker::GetPVRManager().OnSleep();
-
+  if (g_application.m_pPlayer->IsPlaying())
+  {
+    m_lastFileItem.reset(new CFileItem(g_application.CurrentFileItem()));
+    //keep the actual offset instead of store and load it from database
+    m_lastFileItem->m_lStartOffset = static_cast<int>(g_application.GetTime() * 75);
```

Comment:
This is correct and works for all Stack types as well. However two points:
1. for the ISO stacks you also need to set m_lastFileItem.m_lStartPartNumber to CApplication::m_currentStackPosition (protected member, so you'll have to add a Getter)
2. for the ISO stacks and ISO DVDs you also need to save the player state. Problem is that right now playerState seems to be coming ONLY from a stored bookmark. So you could circumvent this by storing the bookmark while you are in OnSleep(). In that case set m_lStartOffset to STARTOFFSET_RESUME.

Note
- you can determine if you are playing an ISO Stack by calling CApplication::IsPlayingISOStack()
- you can determine if you are playing an ISO FILE by checking CurrentFileItem().IsDiscImage()
- you need my recent proposed changes re. stacking (PR #13235)

Reply:
Thanks, that would be fixed in next step.